### DISCUSSION TITLE
Best practices / Performance

### DISCUSSION THREAD

**Discussion by karsvaniersel** [CodeDredd/pinia-orm#1837] [Ideas]
Would be really cool to have a section on the docs that goes a bit more into how to query data in the most optimal way.

I already ran into some edge cases that for some reason made the queries "slow", so would be handy to fallback on some "best practices" set of guidelines. 

Of course super low priority! 

Hopefully I will have some spare time + enough knowledge to start writing some initial docs for this  :)

**Comment by CodeDredd** [comment]
Yeah i have the same idea in mind. Would be great if you could help out. 
Documentation is never low prio...i just don't like doing it ^^. It take huge amount of time.
I think i spend 30% of the time only for docs and still not enough.

One  think i can say. Never use `withAllRecursive` ^^.... the way it's implemented can lead to huge performance drops. @vesper8 can tell you what i mean.
